### PR TITLE
Jacob prototype enemy ant

### PR DESCRIPTION
**Pull request by jDenyes**
This adds an enemy ant. The player collides with ant and loses health. Upon reaching 0 health the level is reloaded.

### CODE REVIEW

File: godotproject/Player.gd [hunk 1]
```diff
@@ -174,11 +176,37 @@ func _physics_process(delta):
...
+		if health <= 0:
+			print("Player is now dead")
+			die()
+			
+		#After the player gets hit set the player to invulnerable to damage for 1 second
+		isVulnerable = false
+		
+		#sets up timer to trigger after 1 second and set player to vulnerable again
+		print("Timer has started")
+		var invulnerable_timer = Timer.new()
```
Comment (fheysel) [inline godotproject/Player.gd]: Should this timer object be created each time the enemy takes damage? Or is there a more efficient way to go about it?

File: godotproject/Player.gd [hunk 2]
```diff
@@ -174,11 +176,37 @@ func _physics_process(delta):
...
+		invulnerable_timer.one_shot = true
+		invulnerable_timer.connect("timeout", self, "on_invulnerable_timeout")
+		add_child(invulnerable_timer)
+		invulnerable_timer.start()
```
Comment (fheysel) [inline godotproject/Player.gd]: Also is this an asynchronous timer? Or will it block the code while it polls?

File: godotproject/Player.gd [hunk 3]
```diff
@@ -174,11 +176,37 @@ func _physics_process(delta):
...
+
+func on_invulnerable_timeout():
```
Comment (fheysel) [inline godotproject/Player.gd]: Do we need to delete the timer object here?

File: godotproject/enemy-Ant.gd [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+#func _process(delta):
+#	pass
+
+
+func _on_PlayerDetector_body_entered(body):
+	print("Enemy has hit player")
+	if(body.isVulnerable):
+		print("Calling player takeDamage(damage)")
+		body.takeDamage(attackDamage)
```
Comment (fheysel) [inline godotproject/enemy-Ant.gd]: Make sure this signal is documented at the bottom of the game plan doc

File: godotproject/project.godot [hunk 5]
```diff
@@ -72,8 +72,6 @@ tongue={
 2d_physics/layer_2="player"
 2d_physics/layer_3="enemy"
 2d_physics/layer_4="tongueable"
-2d_physics/layer_5="spikes"
-2d_physics/layer_6="star"
```
Comment (fheysel) [inline godotproject/project.godot:76]: Why did you delete my physics layers? Does the star collection still work like this?

**Review by fheysel — CHANGES_REQUESTED** [review]
Looks good for the most part. Im not familiar with timers so I had some questions about that. Also were the star and spike collision layers intentionally deleted? Will the star collection still work?

File: godotproject/project.godot (hunk 5 above)
Comment (jDenyes) [reply godotproject/project.godot:76]: mmm, I'm not sure how these got deleted. Definitely was not intentional, we can add it back in

**Review by fheysel — APPROVED** [review]

**Comment by fheysel** [comment]
Whats the status of this pr @jDenyes ?

**Comment by charasyn** [comment]
I think this branch is missing the changes Jacob and I made together to get the audio working. @jDenyes can you see if you can commit those and then we can review?
